sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java

```diff
@@ -93,7 +94,8 @@
    * No two instances of this class are considered equivalent hence we generate a random hash code
    * between 0 and {@link Integer#MAX_VALUE}.
    */
-  private final int hashCode = (int) (Math.random() * Integer.MAX_VALUE);
+  private final int hashCode = (RANDOM.nextInt() * Integer.MAX_VALUE);
```

Comment:
Please swap to use ThreadLocalRandom.current().nextInt(Integer.MIN_VALUE, Integer.MAX_VALUE) instead of the static RANDOM

https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ThreadLocalRandom.html